graphql/src/test/java/com/linecorp/armeria/server/graphql/ExecutionIdGeneratorTest.java

```diff
@@ -0,0 +1,130 @@
+                                                          ExecutionStrategyParameters parameters) {
+            executionId = executionContext.getExecutionId();
+            return super.execute(executionContext, parameters);
+        }
+    }
+
+    static class ConcatExecutionIdGenerator implements ExecutionIdGenerator {
+        @Override
+        public ExecutionId generate(ServiceRequestContext ctx, String query, String operationName) {
+            return ExecutionId.from(ctx + query + operationName);
+        }
+    }
+
+    @RegisterExtension
+    static ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) throws Exception {
+            final File graphqlSchemaFile =
+                    new File(getClass().getResource("/test.graphqls").toURI());
+            final GraphqlService service =
+                    GraphqlService.builder()
+                                  .configureGraphql(builder -> builder.queryExecutionStrategy(idStrategy))
+                                  .schemaFile(graphqlSchemaFile)
+                                  .runtimeWiring(c -> {
+                                      final DataFetcher<String> bar = dataFetcher("bar");
+                                      c.type("Query",
+                                             typeWiring -> typeWiring.dataFetcher("foo", bar));
+                                  })
+                                  .build();
+            final GraphqlService customizedExecutionIdService =
+                    GraphqlService.builder()
+                                  .executionIdGenerator(new ConcatExecutionIdGenerator())
+                                  .configureGraphql(builder -> builder.queryExecutionStrategy(idStrategy))
+                                  .schemaFile(graphqlSchemaFile)
+                                  .runtimeWiring(c -> {
+                                      final DataFetcher<String> bar = dataFetcher("bar");
+                                      c.type("Query",
+                                             typeWiring -> typeWiring.dataFetcher("foo", bar));
+                                  })
+                                  .build();
+            sb.service("/graphql", service)
+              .service("/graphql-customized-execution-id", customizedExecutionIdService);
+        }
+    };
+
+    static DataFetcher<String> dataFetcher(String value) {
+        return environment -> {
+            final ServiceRequestContext ctx = GraphqlServiceContexts.get(environment);
+            assertThat(ctx.eventLoop().inEventLoop()).isTrue();
+            // Make sure that a ServiceRequestContext is available
+            assertThat(ServiceRequestContext.current()).isSameAs(ctx);
+            return value;
+        };
+    }
+
+    @Test
+    void defaultExecutionIdGenerator() throws InterruptedException {
+        BlockingWebClient.of(server.httpUri()).get("/graphql?query={foo}");
```

Comment:
Should we check the response body to ensure the query is performed correctly?
```suggestion
        server.blockingWebClient().get("/graphql?query={foo}");
```